### PR TITLE
feat: added textures to crops and lighting

### PR DESCRIPTION
**Pull request by wifimug**
- added light controller
- added crop textures
- added sky background

### CODE REVIEW

**Comment by DanielSwarup** [comment]
For future note, the two features should have been two separate PRs

File: src/controllers/crop_controller.py [hunk 1]
```diff
@@ -48,7 +49,8 @@ def setup_crops(self):
             crop_model = self.add_crop(crop_size, loc, locx)
             material, segmentation_id = self.assign_crop_type(self.type[curr_crop_type])
 
-            crop_model.active_material = material
+
+            # crop_model.active_material = material
```
Comment (DanielSwarup) [inline src/controllers/crop_controller.py:53]: If not important, remove commented out code

File: src/controllers/crop_controller.py [hunk 2]
```diff
@@ -59,16 +61,22 @@ def assign_crop_type(self, crop_type):
         segmentation_id = 0
         if crop_type == "red":
             material = bpy.data.materials.new("Red")
-            material.diffuse_color = (1,0,0,0.8)
+            # material.diffuse_color = (1,0,0,0.8)
```
Comment (DanielSwarup) [inline src/controllers/crop_controller.py:64]: Same here

File: src/controllers/crop_controller.py [hunk 3]
```diff
@@ -59,16 +61,22 @@ def assign_crop_type(self, crop_type):
...
             segmentation_id = 1
         elif crop_type == "green":
             material = bpy.data.materials.new("Green")
-            material.diffuse_color = (0,1,0,0.8)
+            # material.diffuse_color = (0,1,0,0.8)
```
Comment (DanielSwarup) [inline src/controllers/crop_controller.py:68]: And here

File: src/controllers/crop_controller.py [hunk 4]
```diff
@@ -59,16 +61,22 @@ def assign_crop_type(self, crop_type):
...
             segmentation_id = 2
         elif crop_type == "blue":
             material = bpy.data.materials.new("Blue")
-            material.diffuse_color = (0,0,1,0.8)
+            # material.diffuse_color = (0,0,1,0.8)
```
Comment (DanielSwarup) [inline src/controllers/crop_controller.py:72]: And here

File: src/launch.py [hunk 5]
```diff
@@ -28,14 +29,18 @@ def launch(config):
...
         cameracon = CameraController()
+        lightcon = LightController()
+        lightcon.add_light()
         cropcon = CropController(config["crop"], collection)
         scenerender = SceneRenderer(config["outfile"][0], collection)
         cameracon.setup_camera("camera_one", (10,0,0), (1.57057,0.00174533,1.57057), "Collection")
         cropcon.setup_crops()
         collection1 = bpy.data.collections.get("Collection")
         dupe = collection1.objects.get("stage11.1")
+        # dupe.hide_set(True)
```
Comment (DanielSwarup) [inline src/launch.py:40]: Same here

**Comment by wifimug** [comment]
> For future note, the two features should have been two separate PRs

Yes, initially it was only meant to be texture but I couldn't see it applied without the lighting. Should I split it anyways?

**Review by DanielSwarup — CHANGES_REQUESTED** [review]
Minor changes with commented-out code. Looking forward to Unit tests.

In the future, let's have one feat + its unit test in a single PR.

**Comment by wifimug** [comment]
Accidentally reset my commit and then committed a merge conflict merge. That's why there's two there.

**Comment by DanielSwarup** [comment]
> Accidentally reset my commit and then committed a merge conflict merge. That's why there's two there.

Would recommend performing a squash merge when merging

**Pull request closed without merging**